### DISCUSSION TITLE
[FAQ] I'm using Velero in multiple clusters. Should I use the same bucket to store all of my backups?

### DISCUSSION THREAD

**Discussion by dsu-igeek** [vmware-tanzu/velero#3370] [Community support Q&A] [answered]
I'm using Velero in multiple clusters. Should I use the same bucket to store all of my backups?

**Comment by dsu-igeek** [comment, marked as answer]
We strongly recommend that each Velero instance use a distinct bucket/prefix combination to store backups. Having multiple Velero instances write backups to the same bucket/prefix combination can lead to numerous problems - failed backups, overwritten backups, inadvertently deleted backups, etc., all of which can be avoided by using a separate bucket + prefix per Velero instance.

It's fine to have multiple Velero instances back up to the same bucket if each instance uses its own prefix within the bucket. This can be configured in your BackupStorageLocation, by setting the spec.objectStorage.prefix field. It's also fine to use a distinct bucket for each Velero instance, and not to use prefixes at all.

Related to this, if you need to restore a backup that was created in cluster A into cluster B, you may configure cluster B with a backup storage location that points to cluster A's bucket/prefix. If you do this, you should configure the storage location pointing to cluster A's bucket/prefix in ReadOnly mode via the --access-mode=ReadOnly flag on the velero backup-location create command. This will ensure no new backups are created from Cluster B in Cluster A's bucket/prefix, and no existing backups are deleted or overwritten.

**Reply by mikephung** [reply]
this is exactly what i did on my case, but in the ReadOnly mode, so i ended up to have two backup-location, 1 readonly and the other ReadWrite mode from wheren  i was able to get all backup using velero get backup.
How can i get all backup from the readonly backup storage location? Currently i can not return the backup list from cluster A

cc: @carlisia 
```
NAME       PROVIDER   BUCKET/PREFIX       PHASE       LAST VALIDATED                   ACCESS MODE   DEFAULT
clusterA   aws        s3-backup/clusterA   Available   2023-07-13 21:30:09 +0200 CEST   ReadOnly      true
clusterB   aws        s3-backup/clusterB  Available   2023-07-13 21:30:09 +0200 CEST   ReadWrite     true
```

**Reply by draghuram** [reply]
Similar issue was reported in https://github.com/vmware-tanzu/velero/issues/6422. Are you using object locks or any similar policies on your buckets?

**Reply by mikephung** [reply]
thanks @draghuram, i think my situation looks differents:

on clusterB, the backup-location clusterB (backup-location-clusterB.yaml), using Bucket bucketA to save,  is in used, it always have been, and listing backup is working. 
on clusterA, the backup-location clusterA (backup-location-clusterA.yaml) which using bucketB as default, with access mode ReadWrite, and backup listing is also working.
Now i use the same backup-location on bucketA (backup-location-clusterA-readonly.yaml), edit access mode to ReadOnly and deployed into clusterB.
Expected result: On clusterB the velero get backup should list all backup that have been saved on both clusterA and clusterB, but it is not working.
On the https://github.com/vmware-tanzu/velero/issues/6422 they mentioned about old backup is no longer working, new backup is working, but in my case, backup on each cluster is working, but the backup readonly which has been configured lastly is not listing.

**Reply by mikephung** [reply]
for your question: there are no lifecycle policies (which can include object lock policies) set on the bucket
